src/App.tsx

```diff
@@ -15,33 +15,105 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  NONE,
+  ALPABET,
+  LENGTH,
+}
+
+export const App: React.FC = () => {
+  const [isStarted, setStarted] = useState(false);
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReset, setReset] = useState(false);
+
+  function getReorderedGoods(
+    goods: string[],
+  ) {
+    // Not to mutate the original array
+    const visibleGoods = [...goods];
+    let x = isReversed;
+    let y = sortType;
+
+    if (isReset) {
+      x = false;
+      y = SortType.NONE;
+    }
+
+    visibleGoods.sort((a, b) => {
+      switch (y) {
+        case SortType.ALPABET:
+          return a.localeCompare(b);
+        case SortType.LENGTH:
+          return a.length - b.length;
+        default:
+          return 0;
+      }
+    });
+
+    if (x) {
+      visibleGoods.reverse();
+    }
+    // Sort and reverse goods if needed
+    // ...
+
+    return visibleGoods;
+  }
+
+  const reorderList = getReorderedGoods(
+    goodsFromServer,
+  );
+
+  return (
+    <div className="App">
+      {!isStarted && (
+        <button
+          type="button"
+          onClick={() => setStarted(!isStarted)}
+        >
+          Start
+        </button>
+      )}
+
+      {isStarted && (
+        <>
+          <button
+            className="button"
+            type="button"
+            onClick={() => setSortType(SortType.ALPABET)}
+          >
+            Sort alphabetically
+          </button>
```

Comment:
add the same rules to all sort btn and reverse

https://www.npmjs.com/package/classnames

```suggestion
          <button
            className={cn("button", {"button--active": sortType === SortType.ALPABET})}
            type="button"
            onClick={() => setSortType(SortType.ALPABET)}
          >
            Sort alphabetically
          </button>
```